src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,465 @@
+        this.xstream = xStreamPersisterFactory.createXMLPersister();
+        if (backupFacade.getRestoreExecutions() != null
+                && !backupFacade.getRestoreExecutions().isEmpty()
+                && backupFacade.getRestoreExecutions().containsKey(jobExecution.getId())) {
+            this.currentJobExecution = backupFacade.getRestoreExecutions().get(jobExecution.getId());
+            this.catalog = backupFacade.getRestoreExecutions().get(jobExecution.getId()).getRestoreCatalog();
+            this.isNew = true;
+        } else {
+            this.currentJobExecution = backupFacade.getBackupExecutions().get(jobExecution.getId());
+            this.catalog = backupFacade.getCatalog();
+            this.xstream.setExcludeIds();
+            this.isNew = false;
+        }
+
+        Assert.notNull(catalog, "catalog must be set");
+
+        this.xstream.setCatalog(this.catalog);
+        this.xstream.setReferenceByName(true);
+        this.xp = this.xstream.getXStream();
+
+        Assert.notNull(this.xp, "xStream persister should not be NULL");
+        
+        JobParameters jobParameters = this.currentJobExecution.getJobParameters();
+        
+        this.dryRun = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_DRY_RUN_MODE, "false"));
+        this.bestEffort = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_BEST_EFFORT_MODE, "false"));
+
+        return doExecute(contribution, chunkContext, jobExecution);
+    }
+    
+    /**
+     * 
+     * @param contribution
+     * @param chunkContext
+     * @param jobExecution 
+     * @return
+     * @throws Exception
+     */
+    abstract RepeatStatus doExecute(StepContribution contribution, ChunkContext chunkContext, JobExecution jobExecution)
+            throws Exception;
+
+    /**
+     * @param targetBackupFolder
+     * @throws IOException
+     */
+    public void backupGWCSettings(Resource targetBackupFolder) {
+        GWCConfigPersister gwcGeoServerConfigPersister = 
+                (GWCConfigPersister) GeoServerExtensions.bean("gwcGeoServervConfigPersister");
+        
+        GWCConfigPersister testGWCCP = 
+                new GWCConfigPersister(xStreamPersisterFactory, new GeoServerResourceLoader(targetBackupFolder.dir()));
+        
+        // Test that everything went well
+        try {
+            testGWCCP.save(gwcGeoServerConfigPersister.getConfig());
+
+            GWCConfig gwcConfig = testGWCCP.getConfig();
+            
+            Assert.notNull(gwcConfig);
+            
+            // TODO: perform more tests and integrity checks on reloaded configuration
+            
+            // Store GWC Providers Configurations
+            Resource targetGWCProviderBackupDir = 
+                    BackupUtils.dir(targetBackupFolder, GeoserverXMLResourceProvider.DEFAULT_CONFIGURATION_DIR_NAME);
+
+            for(GeoserverXMLResourceProvider gwcProvider : GeoServerExtensions.extensions(GeoserverXMLResourceProvider.class)) {
```

Comment:
Roger! Given also the other comments, I guess I need to review and improve a bit the backup of the security sub-system.